Library/SharedFunctions.swift

```diff
@@ -388,3 +388,94 @@ public func isEndDateAfterToday(for reward: Reward) -> Bool {
   return (reward.endsAt == nil || (reward.endsAt ?? 0) >= AppEnvironment.current.dateType.init()
     .timeIntervalSince1970)
 }
+
+/*
+ A helper that assists in rounding an amount to a given number of decimal places
+ */
+public func rounded(_ value: Double, places: Int) -> Double {
+  let divisor = pow(10.0, Double(places))
+  return (value * divisor).rounded() / divisor
+}
+
+public func checkoutProperties(
+  from project: Project,
+  baseReward: Reward,
+  rewards: [Reward],
```

Comment:
I think rewards are a property of `project` or `project.rewardData` so I would double check if this parameter is necessary.

Reply:
I think it represents selected rewards not all the project rewards, but I'm not so sure about that as well. I've seen cases in the project where a single selected reward was turned into an array of one item.